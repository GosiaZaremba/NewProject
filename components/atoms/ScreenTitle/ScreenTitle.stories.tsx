import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { ScreenTitle } from './ScreenTitle';

storiesOf('atoms/ScreenTitle', module)
    .addDecorator((story) => story())
    .add('default', () => <ScreenTitle screenTitle={'Screen Title'} />);
