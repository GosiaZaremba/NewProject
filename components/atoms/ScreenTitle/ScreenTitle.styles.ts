import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  titleText: {
    fontSize: 30,
  },
});
