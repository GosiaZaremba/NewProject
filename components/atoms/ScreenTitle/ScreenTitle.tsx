import React from 'react';
import { Text } from 'react-native';
import styles from './ScreenTitle.styles';

export type Props = {
    screenTitle: string;
    testID?: string;
};

export const ScreenTitle: React.FC<Props> = ({ screenTitle, testID }) => {
    return (
        <Text style={styles.titleText} testID={testID}>
            {screenTitle}
        </Text>
    );
};
