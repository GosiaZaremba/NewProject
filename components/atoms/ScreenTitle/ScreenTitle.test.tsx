import { render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import { ScreenTitle } from './ScreenTitle';

describe('Atom ScreenTitle', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <ScreenTitle
                    screenTitle={'ScreenTtile text'}
                    testID={'atom-screenTtile'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <ScreenTitle
                screenTitle={'ScreenTtile text'}
                testID={'atom-screenTtile'}
            />
        );

        const screenTitle = getByTestId('atom-screenTtile');
        expect(screenTitle).toHaveTextContent('ScreenTtile text');
    });
});
