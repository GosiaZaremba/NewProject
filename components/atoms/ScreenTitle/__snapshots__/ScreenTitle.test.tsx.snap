// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom ScreenTitle Component Snapshot 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "fontSize": 20,
      "fontWeight": "700",
    }
  }
  testID="atom-screenTtile"
>
  ScreenTtile text
</Text>
`;
